growthbook/growthbook
# Fix operator precedence in advanced attribute targeting

### Features and Changes

Currently due to the use of `in`, a single special operator (or/nor/and/not) in a nested `ConditionInterface` will ignore all other keys at that level. From the [mongo docs](https://www.mongodb.com/docs/manual/tutorial/query-documents/#specify-and-as-well-as-or-conditions) it looks like we should be allowing multiple keys on a single level, even with operators like `$or`.

### Testing

1. Create a feature with an advanced targeting rule that has at least one `$or` (or other special operator) and at least one other operator, both at the top level.
2. Set up attributes such that the `$or` condition passes, but the other condition fails
3. Observe that the user is still exposed to the experiment without this change, but is no longer exposed with it

### Screenshots
Experiment setup:
![image](https://github.com/growthbook/growthbook/assets/10674248/8ba5dce6-fe9c-4345-94c8-466801164be6)


#### Importing from npm package (no fix)
![image](https://github.com/growthbook/growthbook/assets/10674248/281b0d0e-139b-499b-aa91-c5b2d75bff38)


#### Importing from local package (with fix)

![image](https://github.com/growthbook/growthbook/assets/10674248/a45a279c-94b9-4f5c-9ee8-d9011f918e38)
![image](https://github.com/growthbook/growthbook/assets/10674248/f4f61010-7bd5-4f9c-a0ec-350ea43fb3a8)
